### PR TITLE
[VGNWB-273] Relation template CRUD views

### PR DESCRIPTION
**Pull request by nowke**
Views included

- List templates
- Create new template
- Update existing template
- Delete existing template

Related Backend - https://github.com/diging/vogon-web/pull/158

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,9 @@
     "test:unit": "vue-cli-service test:unit"
   },
   "dependencies": {
+    "@types/clone-deep": "^4.0.1",
     "@types/jwt-decode": "^2.2.1",
+    "clone-deep": "^4.0.1",
```
Comment (tjquinn1) [inline package.json]: For the sake of avoiding small projects lets swap this out for lodash. I believe that the clone deep is part of the lodash core package which shouldn't add too much size to our bundle.

File: src/constants.ts [hunk 2]
```diff
@@ -10,3 +10,20 @@ export const CONCEPT_STATES = {
 };
 
 export const PAGE_SIZE = 20;
+
+export const TEMPLATE_RELATION_TYPES = {
```
Comment (tjquinn1) [inline src/constants.ts:14]: Were these types hard coded before? What if we want to add a new type?

**Review by tjquinn1 — CHANGES_REQUESTED** [review]

File: src/constants.ts (hunk 2 above)
Comment (nowke) [reply src/constants.ts:14]: These are hardcoded in backend also(models, forms) for rendering in templates. If we need to add new type, it requires code change, handling cases, etc. anyways from backend.

File: package.json (hunk 1 above)
Comment (nowke) [reply package.json]: Yeah. That's done

**Comment by nowke** [comment]
Resolved comments

**Review by tjquinn1 — APPROVED** [review]

**Pull request merged**